### PR TITLE
Feat: [CHK-2525] Additional payment methods

### PR DESCRIPTION
**Pull request by mmoio**
#### List of Changes
- Returned payment methods are no longer filtered
   against the PaymentMethodRoutes object
- Fires a mixpanel event on method choice
- Removes fronted side typing extending the generated model
- Refactors sorting and normalization of the received PayementMethods

#### Motivation and Context
This pr makes so all the received payment methods are rendered and presented to the user, keeps the filtering out of duplicated methods. Please note that duplicated methods are not used and while filtering those out the method status is not taken in consideration.

#### How Has This Been Tested?
Visual testing using the [checkout-be-mock](https://github.com/pagopa/pagopa-checkout-be-mock) on branch [CHK-2525--addmore-paymentmethod](https://github.com/pagopa/pagopa-checkout-be-mock/pull/66)

#### Screenshots (if appropriate):
image
image

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/features/payment/components/PaymentChoice/PaymentChoice.tsx [hunk 1]
```diff
@@ -1,110 +1,104 @@
-/* eslint-disable sonarjs/cognitive-complexity */
-/* eslint-disable @typescript-eslint/no-unused-expressions */
-/* eslint-disable functional/immutable-data */
 import React from "react";
+import mixpanel from "mixpanel-browser";
+import { PAYMENT_METHODS_CHOICE } from "utils/config/mixpanelDefs";
```
Comment (vito80ba) [inline src/features/payment/components/PaymentChoice/PaymentChoice.tsx:3]: FIX with a relative import like the other ones 
`../../../../utils/config/mixpanelDefs`

Comment (mmoio) [reply src/features/payment/components/PaymentChoice/PaymentChoice.tsx]: Sorry, I'have amended the last commit fixing the import

**Review by vito80ba — APPROVED** [review]

File: src/features/payment/components/PaymentChoice/PaymentChoice.tsx [hunk 2]
```diff
@@ -1,110 +1,112 @@
...
-  methodCP && sortedMethods.unshift(methodCP);
-
-  return sortedMethods;
-}
+  const { enabled, disabled } = methods.reduce<{
+    enabled: Array<PaymentInstrumentsType>;
+    disabled: Array<PaymentInstrumentsType>;
+  }>(
+    (acc, method) =>
+      method.status === "ENABLED"
```
Comment (Fernando-Granato) [inline src/features/payment/components/PaymentChoice/PaymentChoice.tsx:57]: The ecommerce generated code contains a StatusEnum for not use a String "ENABLED"?

**Review by Fernando-Granato — COMMENTED** [review]
Just a little considerations

File: src/features/payment/components/PaymentChoice/PaymentChoice.tsx [hunk 3]
```diff
@@ -1,110 +1,112 @@
...
       ) : (
         <>
-          <EnabledPaymentMethods
-            methods={getPaymentMethods()}
+          <MethodComponentList
+            methods={paymentMethods.enabled}
             onClick={handleClickOnMethod}
+            testable
           />
-          <DisabledPaymentMethods methods={getPaymentMethods("DISABLED")} />
```
Comment (Fernando-Granato) [inline src/features/payment/components/PaymentChoice/PaymentChoice.tsx:107]: The same for this. "DISABLED"

File: src/features/payment/components/PaymentChoice/PaymentChoice.tsx (hunk 2 above)
Comment (mmoio) [reply src/features/payment/components/PaymentChoice/PaymentChoice.tsx]: You are right, thanks. Fixed with the latest commit

File: src/features/payment/components/PaymentChoice/PaymentChoice.tsx (hunk 3 above)
Comment (mmoio) [reply src/features/payment/components/PaymentChoice/PaymentChoice.tsx:107]: Fixed with the latest commit

**Pull request merged**